### PR TITLE
Add a blocked option for train_test_split

### PR DESCRIPTION
**Pull request by leouieda**
Add option `blocked=False` to `verde.train_test_split` that uses `BlockShuffleSplit` if `True` instead of the scikit-learn `ShuffleSplit`. Also add a gallery example comparing the two options. 

**Reminders**:

- [x] Run `make format` and `make check` to make sure the code follows the style guide.
- [x] Add tests for new features or tests that would have caught the bug that you're fixing.
- [x] Add new public functions/methods/classes to `doc/api/index.rst` and the base `__init__.py` file for the package.
- [x] Write detailed docstrings for all functions/classes/methods. It often helps to design better code if you write the docstrings first.
- [x] If adding new functionality, add an example to the docstring, gallery, and/or tutorials.
- [x] Add your full name, affiliation, and [ORCID](https://orcid.org) (optional) to the `AUTHORS.md` file (if you haven't already) in case you'd like to be listed as an author on the [Zenodo](https://zenodo.org/communities/fatiando) archive of the next release.

### CODE REVIEW

File: examples/train_test_split.py [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+import matplotlib.pyplot as plt
+import cartopy.crs as ccrs
+import verde as vd
+
+# Let's split the Baja California shipborne bathymetry data
+data = vd.datasets.fetch_baja_bathymetry()
+coordinates = (data.longitude, data.latitude)
+values = data.bathymetry_m
+
+# Assign 30% of the data to the testing set.
```
Comment (santisoler) [inline examples/train_test_split.py:16]:
```suggestion
# Assign 20% of the data to the testing set.
```

**Comment by leouieda** [comment]
Thanks @santisoler! Done
